dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -266,6 +267,11 @@ public static LoopStart newLoopStart() {
         public static LoopEnd newLoopEnd() {
             return new LoopEnd();
         }
+
+        public static LoopBound newLoopBound(int bound) {
+            Preconditions.checkArgument(bound >= 0, "Cannot create LoopBound event with negative bound %s", bound);
```

Comment:
`Bound==0` doesn't make much sense for unrolling(*), so it will fall back to the default unrolling.
My reasoning was this: As a user, it would be handy to be able to disable the annotation by specifying some non-sensical bound. This makes sense when setting the bounds via a macro `#define myBound B`.
The question is whether all non-sensical values (i.e. also negative ones) should fall back to the default.

(*) There might be rare cases where you want to have the loop deleted (reduced to a single `goto END_T  #BOUND`). If you expect the code to be dead or something like that.

Reply:
(*) is why I asked if this would kill the code, because I could see an use case for this. But as I understand your other comment, we are not doing this. If the user would want to fallback to the default bound, then he would not use the annotation. I think it makes much more sense to allow bound=0 but use it to kill the loop.

Reply:
But what exactly means killing the loop? Just generating the bound event? I'm not really sure where you would use this practically. I guess you could abuse it to mark dead code by adding a loop and unroll it 0 times, so that the subsequenet code becomes unreachable.

Regarding the fallback: I think if you have macro-heavy code, you would do
```
#ifdef VERIFICATION
   __VERIFIER_loop_bound(BOUND);
```
rather than
```
#ifdef VERIFICATION && USECUSTOMBOUND
   __VERIFIER_loop_bound(BOUND);
```
In the later case you need another macro definition for something that can be more easily achieved by setting the bound to some special value.
If we want to have `bound==0` for killing loops, I would just make `bound<0` result in a NOOP.

Reply:
I opted for having all bounds `<= 0` to result in the default unrolling. The reason why I didn't implement `bound=0` to cause loop deletion is because it requires special logic that doesn't fit into the unrolling code well.
Since the effect of loop deletion basically boils down to just making the loop and all following code unreachable (dead), it would make more sense to add a `__VERIFIER_unreachable` event that does exactly that without relying on loops. It would even be easier to implement cause its compilation should just result in a `goto END_OF_T #BOUND` event.